Platform/Authentication/Token Provider/DefaultTokenProvider.swift

```diff
@@ -0,0 +1,228 @@
+    ///     supply any body items which your endpoint might require to process the request. If not
+    ///     specified otherwise, the token provider will always add grant_type=client_credentials item
+    ///     to the body of the request.
+    ///     - logger: An optional logger used by the token provider.
+    public init(url: URL, headers: [String : String]? = nil, queryItems: [URLQueryItem]? = nil, body: [URLEncodedBodyItem]? = nil, logger: PPLogger? = nil) {
+        self.url = url
+        self.headers = headers
+        self.queryItems = queryItems
+        self.body = body
+        self.headersInjector = nil
+        self.queryItemsInjector = nil
+        self.bodyInjector = nil
+        self.logger = logger
+    }
+    
+    // MARK: - Token retrieval
+    
+    /// Method called by the SDK to authenticate the user.
+    ///
+    /// - Parameters:
+    ///     - completionHandler: The completion handler that provides
+    ///     `AuthenticationResult` to the SDK.
+    public func fetchToken(completionHandler: @escaping (AuthenticationResult) -> Void) {
+        self.updateRequestParametersIfNeeded()
+        
+        do {
+            let request = try self.prepareRequest()
+            
+            self.fetchToken(with: request, completionHandler: completionHandler)
+        } catch {
+            completionHandler(.failure(error: error))
+        }
+    }
+    
+    // MARK: - Private methods
+    
+    private func updateRequestParametersIfNeeded() {
+        if let headersInjector = self.headersInjector {
+            self.headers = headersInjector()
+        }
+        
+        if let queryItemsInjector = self.queryItemsInjector {
+            self.queryItems = queryItemsInjector()
+        }
+        
+        if let bodyInjector = self.bodyInjector {
+            self.body = bodyInjector()
+        }
+    }
+    
+    private func prepareRequest() throws -> URLRequest {
+        let requestURL = try self.requestURL(with: self.url, queryItems: self.queryItems)
+        
+        var request = URLRequest(url: requestURL)
+        request.httpMethod = HTTPMethod.POST.rawValue
+        
+        let headers = self.requestHeaders(for: self.headers)
+        for (field, value) in headers {
+            request.setValue(value, forHTTPHeaderField: field)
+        }
+        
+        let body = self.requestBody(for: self.body)
+        
+        guard let httpBody = URLEncodedBodySerializer.serialize(body).data(using: .utf8) else {
+            self.logger?.log("\(String(describing: self)) failed to build request body.", logLevel: .error)
+            throw AuthenticationError.failedToSerializeBody
+        }
+        
+        request.httpBody = httpBody
+        
+        return request
+    }
+    
+    private func requestURL(with baseURL: URL, queryItems: [URLQueryItem]?) throws -> URL {
+        guard var components = URLComponents(url: self.url, resolvingAgainstBaseURL: false) else {
```

Comment:
This is a copy & past issue. I will fix it in my next commit (it won't affect any functionality). When it come the second part of your comment, I don't see another example of an instance property being used instead of a function parameter.

Reply:
Sorry there was just one other....

https://github.com/pusher/pusher-platform-swift/blob/d83bc9aee62703013b6e26d963ef300ff7c81dc6/Platform/Authentication/Token%20Provider/DefaultTokenProvider.swift#L161

Reply:
Done

Reply:
Did you have any thoughts on my suggestions?